server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -713,12 +730,12 @@ private void cleanupRestoreState(ClusterChangedEvent event) {
 
         RestoreInProgress restoreInProgress = state.custom(RestoreInProgress.TYPE);
         if (restoreInProgress != null) {
-            for (RestoreInProgress.Entry entry : restoreInProgress.entries()) {
-                if (entry.state().completed()) {
-                    assert completed(entry.shards()) : "state says completed but restore entries are not";
+            for (ObjectObjectCursor<String, RestoreInProgress.Entry> entry : restoreInProgress.entries()) {
```

Comment:
can we iterate purely over `values` here, we don't need the key? Might also be convenient if `RestoreInProgress` would implement `Iterable<RestoreInProgress.Entry>`?